lighthouse-core/audits/errors-in-console.js

```diff
@@ -42,11 +44,43 @@ class ErrorLogs extends Audit {
     };
   }
 
+  /** @return {AuditOptions} */
+  static defaultOptions() {
+    return {
+      ignoredPatterns: undefined,
+    };
+  }
+
+
+  /**
+   * @template {{description: string | undefined}} T
+   * @param {Array<T>} items
+   * @param {AuditOptions} options
+   * @return {Array<T>}
+   */
+  static filterAccordingToOptions(items, options) {
+    const {ignoredPatterns} = options;
+    if (!ignoredPatterns) return items;
+
+    return items.filter(({description}) => {
+      if (!description) return true;
+      for (const pattern of ignoredPatterns) {
+        if (pattern instanceof RegExp && pattern.test(description)) return false;
+        if (typeof pattern === 'string' && description.includes(pattern)) return false;
+      }
+
+      return true;
+    });
```

Comment:
what are we going to do with poorly formed `ignoredPatterns`? The error messages aren't going to be very comprehensible (e.g. not an Array) and the silent failure when there isn't an exception (e.g. misspelled or not a string or regex) isn't going to be noticeable

Reply:
`TypeError: ignoredPatterns is not iterable` seems like a quite comprehensible error to fail with, IMO, especially given the sophistication necessary to set audit options at all.

I'm not sure how we could ever possibly know that a user has misspelled the error they want to ignore, so you've lost me on that one. users gotta learn to crawl on their own sometime 😆

Reply:
> `TypeError: ignoredPatterns is not iterable` seems like a quite comprehensible error to fail with, IMO, especially given the sophistication necessary to set audit options at all.

yeah, that does read well

> I'm not sure how we could ever possibly know that a user has misspelled the error they want to ignore, so you've lost me on that one. users gotta learn to crawl on their own sometime 😆

`options: {ignoredPatternzzz: [/s.mple/i, 'really']}` and nothing happens. How do I figure out why?

Reply:
Oooooh misspell the *option name*, that makes much more sense. 👍 

Honestly, I strongly dislike the pattern we've been starting to apply that unknown properties are fatal errors which breaks backwards/forwards compatibility. Would a warning of unrecognized property names address your concern?

Reply:
> Honestly, I strongly dislike the pattern we've been starting to apply that unknown properties are fatal errors which breaks backwards/forwards compatibility. Would a warning of unrecognized property names address your concern?

yeah, sounds great 👍